internal/lock.go

```diff
@@ -165,6 +252,36 @@ func (l *Lock) Download(dir string, tags []string, notags []string, perm string,
 	for i, r := range filteredResources {
 		resource := r
 		go func() {
+			// Try Artifactory first if available
+			if resource.CacheUri != "" {
+				token := os.Getenv("GRABIT_ARTIFACTORY_TOKEN")
+				if token != "" {
+					artifactoryURL := fmt.Sprintf("%s/%s", resource.CacheUri, resource.Integrity)
+					filename := resource.Filename
+					if filename == "" {
+						filename = path.Base(resource.Urls[0])
+					}
+					fullPath := filepath.Join(dir, filename)
+
+					// Use getUrl with bearer token
+					tmpPath, err := getUrl(artifactoryURL, fullPath, token, ctx)
+					if err == nil {
+						if mode != NoFileMode {
+							err = os.Chmod(tmpPath, mode.Perm())
+						}
+						if err == nil {
+							errorCh <- nil
+							if statusLine != nil {
+								statusLine.Increment(i)
+							}
+							return
+						}
+					}
+					if strings.Contains(err.Error(), "lookup invalid") || strings.Contains(err.Error(), "dial tcp") {
+						fmt.Printf("Failed to download from Artifactory, falling back to original URL: %v\n", err)
+					}
```

Comment:
I might be missing something here... but where are you doing the file integrity check?